enterprise/tomcat5/src/org/netbeans/modules/tomcat5/ui/wizard/AddInstanceIterator.java

```diff
@@ -224,14 +224,14 @@ private void checkStartupScript(File homeDir) {
                                 FileUtil.copyFile(
                                     FileUtil.toFileObject(new File(bundledHome, "bin/" + CATALINA)), // NOI18N
                                     FileUtil.toFileObject(new File(homeDir, "bin")),    // NOI18N
-                                    CATALINA.substring(0, CATALINA.indexOf("."))    // NOI18N
+                                    CATALINA.substring(0, CATALINA.indexOf('.'))    // NOI18N
```

Comment:
`indexOf` is the last remaining String method which has the char variant outperforming the single-char-string variant (under some circumstances).

But this is more of a performance bug of the JDK/JVM and highly depends of the length of the string which is being checked (String variant can be sometimes faster! The correct behavior would be identical performance). 

I don't think we should do those kind of refactoring since: 
 - firstly, they should perform the same if the JVM does its job correctly, 
 - secondly, right now, String going to outperform the char variant in some situations, so its a de-optimization ;)
 - thirdly, the difference is so tiny that it would only show up in measurements when used in inner loops without any other bottle necks

We should probably add a note to the description of the code inspection that it is a bit outdated (just like when IDEs suggested the wrong version of toArray() for so long).

I tried to fix it in the JDK but ended up running out of time when I noticed that the initial assumption that char variants are faster was wrong due to another performance bug:
https://github.com/openjdk/jdk/pull/6509
https://github.com/openjdk/jdk/pull/6509#issuecomment-980681069 (string variant faster than char bug)

next winter maybe

Reply:
You are right the difference should be negligible when it is patched.